packages/block-library/src/navigation-link/edit.js

```diff
@@ -99,15 +100,23 @@ const useIsInvalidLink = ( kind, type, id ) => {
 	const isPostType =
 		kind === 'post-type' || type === 'post' || type === 'page';
 	const hasId = Number.isInteger( id );
+	const blockEditingMode = useBlockEditingMode();
+
 	const postStatus = useSelect(
 		( select ) => {
 			if ( ! isPostType ) {
 				return null;
 			}
+
+			// Avoid querying the post status if the block editing mode is disabled.
```

Comment:
```suggestion
			// Fetching the posts status is an "expensive" operation. Especially for sites with large navigations.
			// When the block is rendered in a template or other disabled contexts we can skip this check in order
			// to avoid all these additional requests that don't really add any value in that mode.
```

Sorry for being pedantic here =D Just think that the why here is more important because otherwise the context why we want this is lost immediately.

Reply:
I always use git reference, but I don't mind additional context in comments 👍